### PR TITLE
Sbra 407 3dcorrection datamodule and unet

### PR DESCRIPTION
**Pull request by leonicoletti**
The branch isn't passing the tests due to an unidentified error with the climetlab's API.
All the other tests are passing, with 87%+ coverage.
I open the PR for code review.

### CODE REVIEW

File: weather_forecast/3dcorrection/unets/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+RUN adduser --create-home ${USER} \
+    && echo "${USER} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+# RUN adduser --disabled-password --gecos "" ${USER} \
+    # && echo "${USER} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+USER ${USER}
+WORKDIR ${HOME}
+ENV PATH="${HOME}/.local/bin:${PATH}"
+RUN echo "export PATH=${PATH}:${HOME}/.local/bin" >> ${HOME}/.bashrc
+
+RUN pip install --upgrade pip
```
Comment (A669015) [inline weather_forecast/3dcorrection/unets/Dockerfile:35]: Why not using the same way to install dependencies ? For the other UC, the nox target is used, it avoids to duplicate in the dockerfile the list of the dependencies already listed in the `requirement.txt` file.

File: weather_forecast/3dcorrection/unets/dataproc.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        x_path, y_path = self.reshard(x, y)
+        self.compute_stats(x_path, y_path)
+
+    def download(self) -> xr.DataArray:
+        """Download the data for 3D Correction UC and return an xr.Array."""
+
+        cml.settings.set("cache-directory", self.cached_data_path)
+        cml_ds = cml.load_dataset(
+            'maelstrom-radiation',
+            subset='tier-1',
```
Comment (cbovalo) [inline weather_forecast/3dcorrection/unets/dataproc.py:76]: You should remove this line if timestep and patchstep are provided.

File: weather_forecast/3dcorrection/unets/configs/unet.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+model:
+    class_path: models.LitUnet1D
+    init_args:
+        data_path: data/
```
Comment (cbovalo) [inline weather_forecast/3dcorrection/unets/configs/unet.yaml:16]: It shouldn't be /data/processed ?

Comment (leonicoletti) [reply weather_forecast/3dcorrection/unets/configs/unet.yaml:16]: This isn't forced as the dataproc preprocessed the data on-the-fly, not storing them in a raw format. But for consistency sake, let's call it data/processed :)

**Review by A669015 — APPROVED** [review]

**Review by cbovalo — APPROVED** [review]

**Pull request merged**